service/src/main/java/bio/terra/profile/service/spendreporting/azure/model/mapper/SpendDataMapper.java

```diff
@@ -0,0 +1,94 @@
+package bio.terra.profile.service.spendreporting.azure.model.mapper;
+
+import bio.terra.profile.model.SpendReport;
+import bio.terra.profile.model.SpendReportingAggregation;
+import bio.terra.profile.model.SpendReportingForDateRange;
+import bio.terra.profile.service.spendreporting.azure.model.SpendCategoryType;
+import bio.terra.profile.service.spendreporting.azure.model.SpendData;
+import bio.terra.profile.service.spendreporting.azure.model.SpendDataItem;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SpendDataMapper {
+  public static final String COST_FORMAT = "%.2f";
+
+  public SpendReport mapSpendData(SpendData data) {
+    var spendData =
+        data.getSpendDataItems().stream()
+            .map(
+                sdi ->
+                    buildSpendReportingForDateRange(
+                        sdi.cost(), sdi.currency(), sdi.spendCategoryType()))
+            .collect(Collectors.groupingBy(SpendReportingForDateRange::getCategory))
+            .entrySet()
+            .stream()
+            .map(
+                kvp -> {
+                  var categoryTotalCost =
+                      kvp.getValue().stream()
+                          .mapToDouble(e -> Double.parseDouble(e.getCost()))
+                          .sum();
+                  var currency =
+                      kvp.getValue().stream()
+                          .findFirst()
+                          .map(SpendReportingForDateRange::getCurrency)
+                          .orElse("n/a");
```

Comment:
In what situations is `n/a` a possible value?

Reply:
Technically it should always have value since it is grouped by category. Each category has some items, so the currency always exists.